queue/queueservice.js

```diff
@@ -88,20 +116,135 @@ class QueueService {
     this.deadlineQueueReady = null;
   }
 
+  /////// START OF IMPLEMENTATION WITH AZURE QUEUE NODE
+
+  // Create queue using azure-queue-node client
+  _createQueue(queue) {
+    return new Promise((accept, reject) => {
+      this.client.createQueue(queue, {}, (err, data) => {
+        err ? reject(err) : accept(data)
+      });
+    });
+  }
+
+  // Count messages in queue using azure-queue-node client
+  _countMessages(queue) {
+    return new Promise((accept, reject) => {
+      this.client.countMessages(queue, {}, (err, data) => {
+        err ? reject(err) : accept(data)
+      });
+    });
+  }
+
+  // Put message in queue using azure-queue-node client
+  _putMessage(queue, message, {visibility, ttl}) {
+    return new Promise((accept, reject) => {
+      this.client.putMessage(queue, JSON.stringify(message), {
+        visibilityTimeout:    visibility,
+        messageTTL:           ttl
+      }, (err, data) => { err ? reject(err) : accept(data) });
+    });
+  }
+
+  // Get messages from queue using azure-queue-node client
+  _getMessages(queue, {visibility, count}) {
+    return new Promise((accept, reject) => {
+      this.client.getMessages(queue, {
+        visibilityTimeout:    visibility,
+        maxMessages:          count
+      }, (err, data) => { err ? reject(err) : accept(data) });
+    }).then(messages => {
+      return messages.map(m => {
+        return {
+          payload:  JSON.parse(m.messageText),
+          remove:   () => {
+            return new Promise((accept, reject) => {
+              this.client.deleteMessage(
+                queue, m.messageId, m.popReceipt,
+                {}, (err) => { err ? reject(err) : accept() }
+              );
+            });
+          }
+        };
+      });
+    });
+  }
+
+  /////// START OF IMPLEMENTATION WITH AZURE STORAGE */
+/*
```

Comment:
Good point... We need to use azure-storage for SAS... but otherwise we don't need it... Anyways, it's nice to keep the impl. that works with it around... now that we have it. So we can test with it over time.